### DISCUSSION TITLE
Force user to press enter on search

### DISCUSSION THREAD

**Discussion by PeterSchulzNL** [KevinVandy/material-react-table#806] [Q&A] [answered]
My filter columns are immediately sending a request when the user enters a character. Is it possible to force the user to press the enter key to start a search? THX

**Comment by PeterSchulzNL** [comment]
With more than 10 million records in a table I need to limit the number of queries send to my server. Any ideas how I can hook into global and column filter fields...?

**Comment by PeterSchulzNL** [comment, marked as answer]
Got it! 😇

**Reply by tdanh2k** [reply]
May I ask how you did it?

**Reply by NarenderDhamarlapally** [reply]
how you did it?

**Reply by KevinVandy** [reply]
When/How you call your api is 100% up to how you want to do it in your code. MRT isn't blocking you from doing that. Add an onkeydown in the muiSearchTextFieldProps, listen for 'Enter', and fetch your api or sync/debounce your managed global filter state however you want.
